### PR TITLE
gui-wm/hyprland: include wlr headers

### PR DESCRIPTION
**Pull request by parona-source**
* pkgconfig files already expects this location so no further modification is required.

Bug: https://bugs.gentoo.org/916760

@MrDuartePT @MrRoy

### CODE REVIEW

**Comment by parona-source** [comment]
````
ask@lane ~/sources/gentoo/gui-wm/hyprland $ tree /var/tmp/portage/gui-wm/hyprland-0.31.0/image/
/var/tmp/portage/gui-wm/hyprland-0.31.0/image/
└── usr
    ├── bin
    │   ├── hyprctl
    │   └── Hyprland
    ├── include
    │   └── hyprland
    │       ├── protocols
    │       │   ├── cursor-shape-v1-protocol.h
    │       │   ├── fractional-scale-v1-protocol.h
    │       │   ├── hyprland-global-shortcuts-v1-protocol.h
    │       │   ├── hyprland-toplevel-export-v1-protocol.h
    │       │   ├── idle-protocol.h
    │       │   ├── linux-dmabuf-unstable-v1-protocol.h
    │       │   ├── pointer-constraints-unstable-v1-protocol.h
    │       │   ├── tablet-unstable-v2-protocol.h
    │       │   ├── tearing-control-v1-protocol.h
    │       │   ├── text-input-unstable-v1-protocol.h
    │       │   ├── wlr-foreign-toplevel-management-unstable-v1-protocol.h
    │       │   ├── wlr-layer-shell-unstable-v1-protocol.h
    │       │   ├── wlr-output-power-management-unstable-v1-protocol.h
    │       │   ├── wlr-screencopy-unstable-v1-protocol.h
    │       │   ├── xdg-output-unstable-v1-protocol.h
    │       │   └── xdg-shell-protocol.h
    │       ├── src
    │       │   ├── Compositor.hpp
    │       │   ├── config
    │       │   │   ├── ConfigDataValues.hpp
    │       │   │   ├── ConfigManager.hpp
    │       │   │   └── defaultConfig.hpp
    │       │   ├── debug
    │       │   │   ├── CrashReporter.hpp
    │       │   │   ├── HyprCtl.hpp
    │       │   │   ├── HyprDebugOverlay.hpp
    │       │   │   ├── HyprNotificationOverlay.hpp
    │       │   │   ├── Log.hpp
    │       │   │   └── TracyDefines.hpp
    │       │   ├── defines.hpp
    │       │   ├── events
    │       │   │   └── Events.hpp
    │       │   ├── helpers
    │       │   │   ├── AnimatedVariable.hpp
    │       │   │   ├── BezierCurve.hpp
    │       │   │   ├── Color.hpp
    │       │   │   ├── MiscFunctions.hpp
    │       │   │   ├── Monitor.hpp
    │       │   │   ├── Region.hpp
    │       │   │   ├── Splashes.hpp
    │       │   │   ├── SubsurfaceTree.hpp
    │       │   │   ├── Timer.hpp
    │       │   │   ├── VarList.hpp
    │       │   │   ├── Vector2D.hpp
    │       │   │   ├── Watchdog.hpp
    │       │   │   ├── WLClasses.hpp
    │       │   │   ├── WLListener.hpp
    │       │   │   ├── WLSurface.hpp
    │       │   │   ├── Workspace.hpp
    │       │   │   ├── X11Stubs.hpp
    │       │   │   └── XWaylandStubs.hpp
    │       │   ├── hyprerror
    │       │   │   └── HyprError.hpp
    │       │   ├── includes.hpp
    │       │   ├── init
    │       │   │   └── initHelpers.hpp
    │       │   ├── layout
    │       │   │   ├── DwindleLayout.hpp
    │       │   │   ├── IHyprLayout.hpp
    │       │   │   └── MasterLayout.hpp
    │       │   ├── macros.hpp
    │       │   ├── managers
    │       │   │   ├── AnimationManager.hpp
    │       │   │   ├── EventManager.hpp
    │       │   │   ├── HookSystemManager.hpp
    │       │   │   ├── input
    │       │   │   │   ├── InputManager.hpp
    │       │   │   │   └── InputMethodRelay.hpp
    │       │   │   ├── KeybindManager.hpp
    │       │   │   ├── LayoutManager.hpp
    │       │   │   ├── ProtocolManager.hpp
    │       │   │   ├── SessionLockManager.hpp
    │       │   │   ├── ThreadManager.hpp
    │       │   │   └── XWaylandManager.hpp
    │       │   ├── pch
    │       │   │   └── pch.hpp
    │       │   ├── plugins
    │       │   │   ├── HookSystem.hpp
    │       │   │   ├── PluginAPI.hpp
    │       │   │   └── PluginSystem.hpp
    │       │   ├── protocols
    │       │   │   ├── FractionalScale.hpp
    │       │   │   ├── GlobalShortcuts.hpp
    │       │   │   ├── Screencopy.hpp
    │       │   │   ├── TextInputV1.hpp
    │       │   │   ├── ToplevelExport.hpp
    │       │   │   ├── ToplevelExportWlrFuncs.hpp
    │       │   │   ├── WaylandProtocol.hpp
    │       │   │   └── XDGOutput.hpp
    │       │   ├── render
    │       │   │   ├── decorations
    │       │   │   │   ├── CHyprDropShadowDecoration.hpp
    │       │   │   │   ├── CHyprGroupBarDecoration.hpp
    │       │   │   │   └── IHyprWindowDecoration.hpp
    │       │   │   ├── Framebuffer.hpp
    │       │   │   ├── OpenGL.hpp
    │       │   │   ├── Renderer.hpp
    │       │   │   ├── Shader.hpp
    │       │   │   ├── shaders
    │       │   │   │   ├── Border.hpp
    │       │   │   │   ├── Shadow.hpp
    │       │   │   │   └── Textures.hpp
    │       │   │   ├── Shaders.hpp
    │       │   │   └── Texture.hpp
    │       │   ├── SharedDefs.hpp
    │       │   ├── version.h
    │       │   ├── version.h.in
    │       │   └── Window.hpp
    │       └── wlr
    │           ├── backend
    │           │   ├── drm.h
    │           │   ├── headless.h
    │           │   ├── interface.h
    │           │   ├── libinput.h
    │           │   ├── multi.h
    │           │   ├── session.h
    │           │   └── wayland.h
    │           ├── backend.h
    │           ├── config.h
    │           ├── interfaces
    │           │   ├── wlr_buffer.h
    │           │   ├── wlr_keyboard.h
    │           │   ├── wlr_output.h
    │           │   ├── wlr_pointer.h
    │           │   ├── wlr_switch.h
    │           │   ├── wlr_tablet_pad.h
    │           │   ├── wlr_tablet_tool.h
    │           │   └── wlr_touch.h
    │           ├── render
    │           │   ├── allocator.h
    │           │   ├── dmabuf.h
    │           │   ├── drm_format_set.h
    │           │   ├── egl.h
    │           │   ├── gles2.h
    │           │   ├── interface.h
    │           │   ├── pass.h
    │           │   ├── pixman.h
    │           │   ├── swapchain.h
    │           │   ├── wlr_renderer.h
    │           │   └── wlr_texture.h
    │           ├── types
    │           │   ├── wlr_buffer.h
    │           │   ├── wlr_compositor.h
    │           │   ├── wlr_content_type_v1.h
    │           │   ├── wlr_cursor.h
    │           │   ├── wlr_cursor_shape_v1.h
    │           │   ├── wlr_damage_ring.h
    │           │   ├── wlr_data_control_v1.h
    │           │   ├── wlr_data_device.h
    │           │   ├── wlr_drm.h
    │           │   ├── wlr_drm_lease_v1.h
    │           │   ├── wlr_export_dmabuf_v1.h
    │           │   ├── wlr_foreign_toplevel_management_v1.h
    │           │   ├── wlr_fractional_scale_v1.h
    │           │   ├── wlr_fullscreen_shell_v1.h
    │           │   ├── wlr_gamma_control_v1.h
    │           │   ├── wlr_idle.h
    │           │   ├── wlr_idle_inhibit_v1.h
    │           │   ├── wlr_idle_notify_v1.h
    │           │   ├── wlr_input_device.h
    │           │   ├── wlr_input_inhibitor.h
    │           │   ├── wlr_input_method_v2.h
    │           │   ├── wlr_keyboard_group.h
    │           │   ├── wlr_keyboard.h
    │           │   ├── wlr_keyboard_shortcuts_inhibit_v1.h
    │           │   ├── wlr_layer_shell_v1.h
    │           │   ├── wlr_linux_dmabuf_v1.h
    │           │   ├── wlr_matrix.h
    │           │   ├── wlr_output.h
    │           │   ├── wlr_output_layer.h
    │           │   ├── wlr_output_layout.h
    │           │   ├── wlr_output_management_v1.h
    │           │   ├── wlr_output_power_management_v1.h
    │           │   ├── wlr_pointer_constraints_v1.h
    │           │   ├── wlr_pointer_gestures_v1.h
    │           │   ├── wlr_pointer.h
    │           │   ├── wlr_presentation_time.h
    │           │   ├── wlr_primary_selection.h
    │           │   ├── wlr_primary_selection_v1.h
    │           │   ├── wlr_region.h
    │           │   ├── wlr_relative_pointer_v1.h
    │           │   ├── wlr_scene.h
    │           │   ├── wlr_screencopy_v1.h
    │           │   ├── wlr_seat.h
    │           │   ├── wlr_server_decoration.h
    │           │   ├── wlr_session_lock_v1.h
    │           │   ├── wlr_shm.h
    │           │   ├── wlr_single_pixel_buffer_v1.h
    │           │   ├── wlr_subcompositor.h
    │           │   ├── wlr_switch.h
    │           │   ├── wlr_tablet_pad.h
    │           │   ├── wlr_tablet_tool.h
    │           │   ├── wlr_tablet_v2.h
    │           │   ├── wlr_tearing_control_v1.h
    │           │   ├── wlr_text_input_v3.h
    │           │   ├── wlr_touch.h
    │           │   ├── wlr_viewporter.h
    │           │   ├── wlr_virtual_keyboard_v1.h
    │           │   ├── wlr_virtual_pointer_v1.h
    │           │   ├── wlr_xcursor_manager.h
    │           │   ├── wlr_xdg_activation_v1.h
    │           │   ├── wlr_xdg_decoration_v1.h
    │           │   ├── wlr_xdg_foreign_registry.h
    │           │   ├── wlr_xdg_foreign_v1.h
    │           │   ├── wlr_xdg_foreign_v2.h
    │           │   ├── wlr_xdg_output_v1.h
    │           │   └── wlr_xdg_shell.h
    │           ├── util
    │           │   ├── addon.h
    │           │   ├── box.h
    │           │   ├── edges.h
    │           │   ├── log.h
    │           │   └── region.h
    │           ├── version.h
    │           ├── xcursor.h
    │           └── xwayland
    │               ├── server.h
    │               ├── shell.h
    │               └── xwayland.h
    └── share
        ├── doc
        │   └── hyprland-0.31.0
        │       └── README.md.bz2
        ├── hyprland
        │   ├── hyprland.conf
        │   ├── wall_2K.png
        │   ├── wall_4K.png
        │   ├── wall_8K.png
        │   ├── wall_anime2_2K.png
        │   ├── wall_anime2_4K.png
        │   ├── wall_anime2_8K.png
        │   ├── wall_anime_2K.png
        │   ├── wall_anime_4K.png
        │   └── wall_anime_8K.png
        ├── man
        │   └── man1
        │       ├── hyprctl.1.bz2
        │       └── Hyprland.1.bz2
        ├── pkgconfig
        │   └── hyprland.pc
        ├── wayland-sessions
        │   └── hyprland.desktop
        └── xdg-desktop-portal
            └── hyprland-portals.conf

38 directories, 214 files
```

**Comment by MrDuartePT** [comment]
Look alright I gonna close my PR and text this

**Comment by MrDuartePT** [comment]
@parona-source after recompiling hyprland-plugins, can confirm is working
image

**Comment by parona-source** [comment]
Wrong bug reference. https://bugs.gentoo.org/916760 correct is this. Will fix once I have access to my laptop.

**Comment by MrDuartePT** [comment]
But it working with your ebuild

**Comment by MrDuartePT** [comment]
@parona-source you gonna back ported for 0.30.0 if yes I will also make hyprland-plugins ebuild for 0.30.0

**Comment by parona-source** [comment]
My comment wasn't about it not working, just a metadata problem that will have to be fixed when its merged.

**Comment by parona-source** [comment]
It could be, or 0.30 could be removed. As im keeping the initial revision, backporting it wouldn't do much eitherway.

**Comment by MrDuartePT** [comment]
> It could be, or 0.30 could be removed. As im keeping the initial revision, backporting it wouldn't do much eitherway.

Yha probably better is just remove 0.30 everyone seems to always update hyprland for latest version. At least you come with simple solution. I kinda overthink a bit when making a solution😂.

**Comment by parona-source** [comment]
> > It could be, or 0.30 could be removed. As im keeping the initial revision, backporting it wouldn't do much eitherway.
> 
> Yha probably better is just remove 0.30 everyone seems to always update hyprland for latest version. At least you come with simple solution. I kinda overthink a bit when making a solution😂.

Worst case have to add >= to the plugins in guru. Really not much of a big deal. If it was stable v  unstable then it would be more of a conundrum.

**Comment by pabaptista** [comment]
Just tried out the new ebuild and it worked just fine here. Allowed me to install hyprland-plugins without any issues.

File: gui-wm/hyprland/hyprland-0.31.0-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	meson_src_install --skip-subprojects wlroots
+	meson_src_install --tags devel
+
+	## Wlroots headers are required by hyprland-plugins and the pkgconfig file expects
+	## them to be in /usr/include/hyprland/wlroots, despite this they aren't installed there.
+	# no meson inbuilt way to override includedir per subproject
+	mkdir "${D}"/usr/include/hyprland/wlroots || die
+	mv "${D}"/usr/include/wlr "${D}"/usr/include/hyprland/wlroots || die
+	# devel tag includes wlroots .pc and .a files still
+	rm -rf "${D}"/usr/lib64/ || die
```
Comment (thesamesam) [inline gui-wm/hyprland/hyprland-0.31.0-r1.ebuild:132]: This should surely be $(get_libdir) and ${ED}.

File: gui-wm/hyprland/hyprland-0.31.0-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	)
+
+	meson_src_configure
+}
+
+src_install() {
+	meson_src_install --skip-subprojects wlroots
+	meson_src_install --tags devel
+
+	## Wlroots headers are required by hyprland-plugins and the pkgconfig file expects
```
Comment (thesamesam) [inline gui-wm/hyprland/hyprland-0.31.0-r1.ebuild:126]: I don't like the double comment here, just use #

**Pull request closed without merging**